h5py/_hl/files.py

```diff
@@ -60,6 +60,19 @@ def make_fapl(driver, libver, **kwds):
         plist.set_fapl_core(**kwds)
     elif driver == 'family':
         plist.set_fapl_family(memb_fapl=plist.copy(), **kwds)
+    elif driver == 'split':
+        if 'metafile' in kwds:
+            kwds['metafile'] = os.path.join(kwds['metafile'],r'%s')
+        kwds['metafile'] = kwds.get('metafile','')+'-m.h5'
+        if 'rawfile' in kwds:
+            kwds['rawfile'] = os.path.join(kwds['rawfile'],r'%s')
+        kwds['rawfile'] = kwds.get('rawfile','')+'-r.h5'
+        try:
+           kwds['rawfile'] = kwds['rawfile'].encode(sys.getfilesystemencoding())
+           kwds['metafile'] = kwds['metafile'].encode(sys.getfilesystemencoding())
```

Comment:
Rather than using `.encode`, can you use `encode_filename` from compat.py. It avoids a number of other problems.